### DISCUSSION TITLE
Pending Result Interface for C API - Callback when result ready

### DISCUSSION THREAD

**Discussion by Giorgi** [duckdb/duckdb#9366] [Ideas]
Using the [Pending Result Interface](https://duckdb.org/docs/api/c/api#pending-result-interface) we can execute a query in the background (assuming that `threads > 1`) and need to repeatedly call `duckdb_pending_execute_task` to poll the result until it returns `DUCKDB_PENDING_RESULT_READY` to process the result. It will be useful to add a new function that will accept a user-provided callback which will be invoked by DuckDB when all tasks are executed and result is available.

**Comment by Tishj** [comment]
Would be great if you could make it clear in the title that this concerns the C-API 👍

**Reply by Giorgi** [reply]
Done.
